### PR TITLE
rustc_driver: Allow running the compiler with a FileLoader

### PR DESCRIPTION
**Pull request by aochagavia**
cc @nrc. I chose to implement this in such a way that it doesn't break anything. Please let me know if you want me to change anything.

### CODE REVIEW

File: src/librustc_driver/lib.rs [hunk 1]
```diff
@@ -189,7 +196,12 @@ pub fn run_compiler<'a>(args: &[String],
     };
 
     let cstore = Rc::new(CStore::new(token::get_ident_interner()));
-    let sess = build_session(sopts, input_file_path, descriptions, cstore.clone());
+    let codemap = Rc::new(CodeMap::with_file_loader(loader));
+    let sess = session::build_session_with_codemap(sopts,
+                                          input_file_path,
+                                          descriptions,
+                                          cstore.clone(),
+                                          codemap);
```
Comment (nrc) [inline src/librustc_driver/lib.rs]: nit: argument alignment

**Comment by nrc** [comment]
lgtm, r+ with the style nit addressed.

**Comment by aochagavia** [comment]
@nrc Done :smile:

**Pull request merged**